src/applications/personalization/profile/components/proof-of-veteran-status/ProofOfVeteranStatus.jsx

```diff
@@ -29,12 +30,14 @@ const ProofOfVeteranStatus = ({
     (/iPad|iPhone|iPod/.test(userAgent) && !window.MSStream) ||
     /android/i.test(userAgent);
 
-  const eligibilityMap = serviceHistory.map(
-    service =>
-      service.characterOfDischargeCode
-        ? DISCHARGE_CODE_MAP[service.characterOfDischargeCode].indicator
-        : 'N',
-  );
+  // Veteran is considered eligible if their most recent discharge code is allow-listed
+  const mostRecentDischargeCode =
+    serviceHistory.length > 0
+      ? serviceHistory.at(-1).characterOfDischargeCode
```

Comment:
Question on this logic.
So we went from checking all the service history for a discharge code of "Y" to only checking the last entry, is that correct?

This might conflict with the language in the one alert where it says:

> you must have received an honorable discharge for at least one period of service